crypto/fipsmodule/cpucap/cpu_aarch64_linux.c

```diff
@@ -127,11 +128,13 @@ void OPENSSL_cpuid_setup(void) {
     OPENSSL_armcap_P |= ARMV8_SHA3;
   }
 
-  // Check if the CPU model is Neoverse V1,
-  // which has a wide crypto/SIMD pipeline.
-  OPENSSL_arm_midr = armv8_cpuid_probe();
-  if (MIDR_IS_CPU_MODEL(OPENSSL_arm_midr, ARM_CPU_IMP_ARM, ARM_CPU_PART_V1)) {
-    OPENSSL_armcap_P |= ARMV8_NEOVERSE_V1;
+  if (hwcap & kCPUID) {
```

Comment:
can we please add a comment what this checks?

Reply:
Sure, I had originally omitted writing a comment since similar hardware capabilities were being checked. But it's probably worth calling out that as of Valgrind 3.21 reading from MIDR_EL1 is still as of yet unsupported.